packages/codegen-ui-react/lib/amplify-ui-renderers/collection.ts

```diff
@@ -16,47 +16,184 @@
+              factory.createJsxAttributes([]),
+            ),
+            [
+              factory.createJsxElement(
+                this.renderCollectionOpeningElement(itemsVariableName),
+                [arrowFuncExpr],
+                factory.createJsxClosingElement(factory.createIdentifier(this.component.componentType)),
+              ),
+              factory.createJsxExpression(
+                undefined,
+                factory.createBinaryExpression(
+                  factory.createIdentifier('isApiPagination'),
+                  factory.createToken(ts.SyntaxKind.AmpersandAmpersandToken),
+                  factory.createJsxSelfClosingElement(
+                    factory.createIdentifier('Pagination'),
+                    undefined,
+                    factory.createJsxAttributes([
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('currentPage'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('pageIndex')),
+                      ),
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('totalPages'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('maxViewed')),
+                      ),
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('hasMorePages'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('hasMorePages')),
+                      ),
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('onNext'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('handleNextPage')),
+                      ),
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('onPrevious'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('handlePreviousPage')),
+                      ),
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('onChange'),
+                        factory.createJsxExpression(undefined, factory.createIdentifier('jumpToPage')),
+                      ),
+                    ]),
+                  ),
+                ),
+              ),
+            ],
+            factory.createJsxClosingElement(factory.createIdentifier('div')),
+          )
+        : factory.createJsxElement(
+            this.renderCollectionOpeningElement(itemsVariableName),
+            [arrowFuncExpr],
+            factory.createJsxClosingElement(factory.createIdentifier(this.component.componentType)),
+          );
 
     this.importCollection.addImport('@aws-amplify/ui-react', this.component.componentType);
 
+    if (this.renderConfig.apiConfiguration?.dataApi === 'GraphQL') {
+      this.importCollection.addMappedImport(ImportValue.API);
```

Comment:
Can we update `addMappedImport` to take in `params`? That way we can pass all 3 in at the same time

Reply:
That's a great idea, I'll make the update.